### DISCUSSION TITLE
Update a .tex and a local .bib jointly

### DISCUSSION THREAD

**Discussion by paquiteau** [emacs-citar/citar#818] [Show and tell]
Hi all, 

I have recently been using emacs and citar for writing LaTeX papers. Each paper has a local bibliography file (ideal for collaboration, synchronisation with overleaf/git, potential CI, etc...) 

I had trouble to use citar in this setup, so I came up with this snippet, that basically update jointly the `.tex` (where the citation is inserted) and the local `.bib` (where a full bibtex entry is added). 

```emacs-lisp
(defun citar-insert-tex-bib (citekeys)
"Insert CITEKEYS both as citation key in tex and as bibtex entry"
  (interactive (list (citar-select-refs)))
  (let ((local-bib-file (concat (car LaTeX-auto-bibliography) ".bib"))
        (updated nil))
  ;; insert in latex buffer
    (citar-latex-insert-citation citekeys nil "cite")
  ;; update in local .bib
    (with-temp-buffer
      (insert-file-contents local-bib-file )
      (mapcar (lambda (citekey)
                (goto-char (point-min))
                (unless (re-search-forward (concat "@.*?{" citekey) (point-max) t)
                  (setq updated t)
                  (goto-char (point-max))
                  (citar--insert-bibtex citekey)))
              citekeys)
      (when updated (write-file local-bib-file)))))
```

I am sharing this for anyone who could find it usefull. It's not much, so it can stay happily in my config, if there is any interest for adding this upstream, let me know :slightly_smiling_face:

**Comment by bdarcus** [comment]
Thanks for sharing @paquiteau!

> I had trouble to use citar in this setup, so I came up with this snippet, that basically update jointly the `.tex` (where the citation is inserted) and the local `.bib` (where a full bibtex entry is added).

To clarify, do you mean transferring from the global to the local bib and the document?

So you want to add a reference to a global item, but then make sure it gets added to the local bib?

**Reply by paquiteau** [reply]
> To clarify, do you mean transferring from the global to the local bib and the document?

Exactly.

**Reply by bdarcus** [reply]
Got it.

On whether this makes sense upstream ...

I don't really use local bibs, so one reason we haven't address this.

It does seems like it would make sense, but it would need to somehow be generalized so it works across major modes, in org-cite etc.

I'm not exactly sure how that should best work (maybe a prefix argument and/or some kind of config option?).

If you have interest to discuss something like that, maybe add a FR to the issue tracker, and link to this?

PS - the wiki is another place to put example code.

EDIT: see also this related issue https://github.com/emacs-citar/citar/issues/699
